firebase-ai/src/main/kotlin/com/google/firebase/ai/type/Candidate.kt

```diff
@@ -317,3 +293,203 @@ public class FinishReason private constructor(public val name: String, public va
+ * within the Service Specific Terms).
+ *
+ * @property webSearchQueries The list of web search queries that the model performed to gather the
+ * grounding information. These can be used to allow users to explore the search results themselves.
+ * @property searchEntryPoint Google search entry point for web searches. This contains an HTML/CSS
+ * snippet that **must** be embedded in an app to display a Google Search Entry point for follow-up
+ * web searches related to the model's "Grounded Response".
+ * @property groundingChunks The list of [GroundingChunk] classes. Each chunk represents a piece of
+ * retrieved content that the model used to ground its response.
+ * @property groundingSupports The list of [GroundingSupport] objects. Each object details how
+ * specific segments of the model's response are supported by the `groundingChunks`.
+ */
+public class GroundingMetadata(
+  public val webSearchQueries: List<String>,
+  public val searchEntryPoint: SearchEntryPoint?,
+  public val retrievalQueries: List<String>,
+  @Deprecated("Use groundingChunks instead")
+  public val groundingAttribution: List<GroundingAttribution>,
+  public val groundingChunks: List<GroundingChunk>,
+  public val groundingSupports: List<GroundingSupport>,
+) {
+  @Serializable
+  internal data class Internal(
+    val webSearchQueries: List<String>?,
+    val searchEntryPoint: SearchEntryPoint.Internal?,
+    val retrievalQueries: List<String>?,
+    @Deprecated("Use groundingChunks instead")
+    val groundingAttribution: List<GroundingAttribution.Internal>?,
+    val groundingChunks: List<GroundingChunk.Internal>?,
+    val groundingSupports: List<GroundingSupport.Internal>?,
+  ) {
+    internal fun toPublic() =
+      GroundingMetadata(
+        webSearchQueries = webSearchQueries.orEmpty(),
+        searchEntryPoint = searchEntryPoint?.toPublic(),
+        retrievalQueries = retrievalQueries.orEmpty(),
+        groundingAttribution = groundingAttribution?.map { it.toPublic() }.orEmpty(),
+        groundingChunks = groundingChunks?.map { it.toPublic() }.orEmpty(),
+        groundingSupports = groundingSupports?.map { it.toPublic() }.orEmpty().filterNotNull()
+      )
+  }
+}
+
+/**
+ * Represents a Google Search entry point.
+ *
+ * @property renderedContent An HTML/CSS snippet that can be embedded in your app. To ensure proper
+ * rendering, it's recommended to display this content within a `WebView`.
+ * @property sdkBlob A blob of data for the client SDK to render the search entry point.
+ */
+public class SearchEntryPoint(
+  public val renderedContent: String,
+  public val sdkBlob: String?,
+) {
+  @Serializable
+  internal data class Internal(
+    val renderedContent: String?,
+    val sdkBlob: String?,
+  ) {
+    internal fun toPublic(): SearchEntryPoint {
+      // If rendered content is null, the user must not display the grounded result. If they do,
```

Comment:
For Firebase, the "user" is generally the "end user".  We also talk *to* our devs, not *about* them.  Here, I would probably write:

If rendered content is null, you must not display the grounded result. If you do, then you violate the service terms. To prevent this from happening, throw an exception.

Reply:
This is a comment that will not be a part of the reference documentation, so we are not talking to our devs here. 
I wanted to leave a code comment in case it's unclear to an SDK dev why we're throwing an exception here, since it's a special case.